client/src/redux/save-challenge-saga.js

```diff
@@ -0,0 +1,66 @@
+import { call, takeEvery, put, select } from 'redux-saga/effects';
+import { postSaveChallenge, mapFilesToChallengeFiles } from '../utils/ajax';
+import {
+  challengeDataSelector,
+  challengeMetaSelector
+} from '../templates/Challenges/redux';
+import { createFlashMessage } from '../components/Flash/redux';
+import { challengeTypes } from '../../utils/challenge-types';
+import { FlashMessages } from '../components/Flash/redux/flash-messages';
+import {
+  uniformizeRequestBody,
+  getStringSizeInBytes,
+  bodySizeFits,
+  getFlashMessage
+} from '../utils/challenge-request-helpers';
+import { saveChallengeComplete } from './';
+
+export function* saveChallengeSaga() {
+  const { id, challengeType } = yield select(challengeMetaSelector);
+  const { challengeFiles } = yield select(challengeDataSelector);
+
+  // only allow saving of multiFileCertProject's
+  if (challengeType === challengeTypes.multiFileCertProject) {
+    const body = uniformizeRequestBody({ id, challengeFiles, challengeType });
+    const bodySizeInBytes = getStringSizeInBytes(body);
+
+    if (!bodySizeFits(bodySizeInBytes)) {
+      return yield put(
+        createFlashMessage(
+          getFlashMessage('flash.challenge-save-too-big', bodySizeInBytes)
+        )
```

Comment:
Were this typed, it would throw a type error:

```suggestion
        createFlashMessage({
          type: 'danger',
          message: FlashMessages.ChallengeSaveTooBig,
          variables: { 'max-size': MAX_BODY_SIZE, 'user-size': bodySizeInBytes }
        })
```